influx/templates.go

```diff
@@ -7,16 +7,34 @@ import (
 )
 
 // TemplateReplace replaces templates with values within the query string
-func TemplateReplace(query string, templates []chronograf.TemplateVar) string {
-	replacements := []string{}
-	for _, v := range templates {
-		newVal := v.String()
-		if newVal != "" {
-			replacements = append(replacements, v.Var, newVal)
+func TemplateReplace(query string, templates chronograf.TemplateVars) string {
+	tvarsByPrecedence := make(map[uint]chronograf.TemplateVars, len(templates))
+	maxPrecedence := uint(0)
+	for _, tmp := range templates {
+		precedence := tmp.Precedence()
+		if precedence > maxPrecedence {
+			maxPrecedence = precedence
 		}
+		tvarsByPrecedence[precedence] = append(tvarsByPrecedence[precedence], tmp)
+	}
+
+	replaced := query
+	for prc := uint(0); prc <= maxPrecedence; prc++ {
```

Comment:
This algorithm is pretty specific.  Would you comment it?  Mostly, I'm interested in you commenting why you are checking for an empty newVal, and why you are handling ExecutableVar differently.